dask/dataframe/core.py

```diff
@@ -1510,7 +1510,9 @@ def to_bag(self, index=False):
         return to_bag(self, index)
 
     def _get_numeric_data(self, how='any', subset=None):
-
+        # If unknown dtype, need to infer from head.
+        if not self._known_dtype:
+            self.dtypes
```

Comment:
I'm not sure. If you just need the column names, then there's no reason to trigger computation. In this case we need the column types as well.